src/coreclr/vm/jitinterface.cpp

```diff
@@ -1721,6 +1719,40 @@ bool CEEInfo::isFieldStatic(CORINFO_FIELD_HANDLE fldHnd)
     return res;
 }
 
+int CEEInfo::getArrayOrStringLength(CORINFO_OBJECT_HANDLE objHnd)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int arrLen = -1;
+    JIT_TO_EE_TRANSITION();
+
+    _ASSERT(objHnd != NULL);
+
+    GCX_COOP();
+
+    Object* obj = OBJECTREFToObject(getObjectFromJitHandle(objHnd));
+
+    if (obj->GetMethodTable()->IsArray())
+    {
+        arrLen = ((ArrayBase*)obj)->GetNumComponents();
+    }
+    else if (obj->GetMethodTable()->IsString())
+    {
+        arrLen = ((StringObject*)obj)->GetStringLength();
+    }
+    else
```

Comment:
```suggestion
    OBJECTREF obj = getObjectFromJitHandle(objHnd);

    if (obj->GetMethodTable()->IsArray())
    {
        arrLen = ((BASEARRAYREF)obj)->GetNumComponents();
    }
    else if (obj->GetMethodTable()->IsString())
    {
        arrLen = ((STRINGREF)obj)->GetStringLength();
    }
    else
```
And similar in other places that call `getObjectFromJitHandle`